File: 준영/LV2/오픈채팅방.js
function solution(record) {
  var record_dict = {}
  var answer1 = []
  var answer2 = []
  var answer = []

  for (const i of record) {
    var temp = i.split(' ')
    if (temp[0] === 'Enter') {
      record_dict[temp[1]] = temp[2]
      answer1.push(temp[1])
      answer2.push('님이 들어왔습니다.')
    } else if (temp[0] === 'Leave') {
      answer1.push(temp[1])
      answer2.push('님이 나갔습니다.')
    } else if (temp[0] === 'Change') {
      record_dict[temp[1]] = temp[2]
    }
  }

  for (let i = 0; i < answer1.length; i++) {
    answer.push(record_dict[answer1[i]] + answer2[i])
  }

  return answer
}


record = ["Enter uid1234 Muzi", "Enter uid4567 Prodo","Leave uid1234","Enter uid1234 Prodo","Change uid4567 Ryan"]	

console.log(solution(record))

File: 준영/프로그래머스/LV2/오픈채팅방.py
def solution(record):
    record_dit = {}
    answer1 = []
    answer2 = []
    answer = []

    for reco in record:
        temp = reco.split()
        if temp[0] == 'Enter':
            record_dit[temp[1]] = temp[2]
            answer1.append(temp[1])
            answer2.append('님이 들어왔습니다.')
        elif temp[0] == 'Leave':
            answer1.append(temp[1])
            answer2.append('님이 나갔습니다.')
        elif temp[0] == 'Change':
            record_dit[temp[1]] = temp[2]

    for i in range(len(answer2)):
        answer.append(record_dit[answer1[i]] + answer2[i])
    
    return answer